tests/setup_/_pattoo/test_packages.py

```diff
@@ -63,22 +65,32 @@ def test_install_missing_pip3(self):
 
     def test_install(self):
         """Unittest to test the install function."""
-        # At least one expected package
-        expected_package = 'PattooShared'
-        expected = True
-        with tempfile.TemporaryDirectory() as temp_dir:
-            result = install(ROOT_DIR, temp_dir)
-
-            # Get raw packages in requirements format
-            packages = subprocess.check_output(
-                [sys.executable, '-m', 'pip', 'freeze'])
-
-            # Get packages with versions removed
-            installed_packages = [
-                package.decode().split('==')[
-                    0] for package in packages.split()]
-            result = expected_package in installed_packages
-        self.assertEqual(result, expected)
+        # Test with undefined requirements directory
+        with self.subTest():
+            with self.assertRaises(SystemExit) as cm_:
+                requirements_dir = data.hashstring(str(random()))
+                install(requirements_dir)
+            self.assertEqual(cm_.exception.code, 3)
+
+        # Test with default expected behaviour
+        with self.subTest():
+            # At least one expected package
+            expected_package = 'PattooShared'
+            expected = True
+
+            # Create temporary directory
+            with tempfile.TemporaryDirectory() as temp_dir:
+                result = install(ROOT_DIR, temp_dir)
+
+                # Get raw packages in requirements format
+                packages = subprocess.check_output(
```

Comment:
Why did you use this instead of the runscript function?

Reply:
my bad, I didn't realise that the runscript function also stored the output at first. I just made a modification that uses it instead of check_output